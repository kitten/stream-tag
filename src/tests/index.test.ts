import { createElement } from 'react';
import { renderToNodeStream } from 'react-dom/server';
import streamToString from 'stream-to-string';
import streamTag from '../index';

describe('streamTag', () => {
  it('handles strings-only correctly', () => {
    const stream = streamTag`test`;

    return streamToString(stream).then(string => {
      expect(string).toBe('test');
    });
  });

  it('handles empty templates correctly', () => {
    const stream = streamTag``;

    return streamToString(stream).then(string => {
      expect(string).toBe('');
    });
  });

  it('handles falsy templates correctly', () => {
    const stream = streamTag`${undefined}${null}${false}`;

    return streamToString(stream).then(string => {
      expect(string).toBe('');
    });
  });

  it('handles string interpolations correctly', () => {
    const stream = streamTag`test${'test'}`;

    return streamToString(stream).then(string => {
      expect(string).toBe('testtest');
    });
  });

  it('handles number interpolations correctly', () => {
    const stream = streamTag`${0}test${9}`;

    return streamToString(stream).then(string => {
      expect(string).toBe('0test9');
    });
  });

  it('handles promise interpolations correctly', () => {
    const stream = streamTag`x${Promise.resolve('test')}x`;

    return streamToString(stream).then(string => {
      expect(string).toBe('xtestx');
    });
  });

  it('handles nested streams correctly', () => {
    const stream = streamTag`x${streamTag`test`}x`;

    return streamToString(stream).then(string => {
      expect(string).toBe('xtestx');
    });
  });

  it('handles Buffers correctly', () => {
    const stream = streamTag`x${Buffer.from('test')}x`;

    return streamToString(stream).then(string => {
      expect(string).toBe('xtestx');
    });
  });

  describe('React SSR Integration', () => {
    it('correctly interpolates renderToNodeStream', () => {
      const tree = createElement('div', { className: 'test' }, (
        createElement('h1', {}, 'Hello World!')
      ));

      const stream = streamTag`
        <html>
          <body>
            ${renderToNodeStream(tree)}
          </body>
        </html>
      `;

      return streamToString(stream).then(string => {
        expect(string).toMatchSnapshot();
      });
    });
  });
});
