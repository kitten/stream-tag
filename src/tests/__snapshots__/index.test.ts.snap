// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`streamTag React SSR Integration correctly interpolates renderToNodeStream 1`] = `
"
        <html>
          <body>
            <div class=\\"test\\" data-reactroot=\\"\\"><h1>Hello World!</h1></div>
          </body>
        </html>
      "
`;
